Add CSI Ephemeral inline volume docs (#727)
portworx/pxdocs: content/portworx-install-with-kubernetes/storage-operations/csi/_index.md

## Supported features

The following table shows the core features supported by CSI and which minimum versions of Portworx and Kubernetes they require.

| Feature | Portworx version | Supported Kubernetes version | Alpha Kubernetes version |
| --- | --- | --- | --- |
| [Provision, attach, and mount volumes](/portworx-install-with-kubernetes/storage-operations/csi/#create-and-use-persistent-volumes) | 2.2 | 1.13.12 |  |
| [CSI Snapshots](/portworx-install-with-kubernetes/storage-operations/csi/#take-snapshots-of-csi-enabled-volumes) | 2.2 | 1.16 | 1.13.12 |
| Stork [^1] | 2.2 | 1.13 |  |
| Volume Expansion (resizing) | 2.2 | 1.16 | 1.14 |
| [Ephemeral Inline Volumes](/portworx-install-with-kubernetes/storage-operations/csi/#ephemeral-inline-volumes-with-csi) | 2.5 | 1.16 | 1.15 |

> It is not recommended to use alpha Kubernetes features in production as the API and core functionality is not finalized. Users that adopt alpha features in production may need to perform costly manual upgrades.

Reviewer: ```suggestion
Portworx, Inc. does not recommend that you use alpha Kubernetes features in production as the API and core functionality are not finalized. Users that adopt alpha features in production may need to perform costly manual upgrades.
```

## Supported features

The following table shows the core features supported by CSI and which minimum versions of Portworx and Kubernetes they require.

| Feature | Portworx version | Supported Kubernetes version | Alpha Kubernetes version |
| --- | --- | --- | --- |
| [Provision, attach, and mount volumes](/portworx-install-with-kubernetes/storage-operations/csi/#create-and-use-persistent-volumes) | 2.2 | 1.13.12 |  |
| [CSI Snapshots](/portworx-install-with-kubernetes/storage-operations/csi/#take-snapshots-of-csi-enabled-volumes) | 2.2 | 1.16 | 1.13.12 |
| Stork | 2.2 | 1.13 |  |
| Volume Expansion (resizing) | 2.2 | 1.16 | 1.14 |
| [Ephemeral Inline Volumes](/portworx-install-with-kubernetes/storage-operations/csi/#ephemeral-inline-volumes-with-csi) | 2.5 | 1.16 | 1.15 |

{{<info>}}
**NOTE:**

> * Note that only Stork 2.3.0 or later is supported with CSI.

Reviewer: @ggriffiths the stork footnote wasn't working for me, so I wrapped them both under a note list.

Author: thanks :+1:

      kubectl apply -f clonePVC.yaml
      ```

## Ephemeral Inline Volumes with CSI

> To use an ephemeral inline volume with the Portworx CSI driver, declare it inside of the pod spec as seen below.

Reviewer: Can we present this in steps like we did for the sections above? 
1. enable feature gate
2. create a pod spec, specifying ???
3. apply the spec to create ???

Author: No feature gate is needed, but yes I'll update for steps 2 and 3

Author: Updated with steps.

      kubectl apply -f clonePVC.yaml
      ```

## Ephemeral Inline Volumes with CSI
You can create ephemeral inline volumes with CSI. These volumes are created during pod creation and deleted upon pod teardown.

1. Create a pod spec that uses the Portworx CSI driver, declaring the inline volume as seen below in a YAML file named `ephemeral-volume-pod.yaml`

> ```

Reviewer: ```suggestion
```text
```

kind: Pod
apiVersion: v1
metadata:
  name: csi-inline-volume-app
spec:
  containers:
    - name: my-frontend
      image: busybox
      volumeMounts:
      - mountPath: "/data"
        name: my-csi-volume
      command: [ "sleep", "1000000" ]
  volumes:
    - name: my-csi-volume
      csi:
        driver: pxd.portworx.com
        volumeAttributes:
          size: "2Gi"
```

2. Apply the `ephemeral-volume-pod.yaml`  spec to create the pod with an ephemeral inline volume:

>       ```text

Reviewer: note for previous comment, do it just like this.

      kubectl apply -f ephemeral-volume-pod.yaml
      ```

> Ephemeral inline volumes may also be used with deployments and other application types. See this [Kubernetes blog](https://kubernetes.io/blog/2020/01/21/csi-ephemeral-inline-volumes/) for more information and example use cases.

Reviewer: ```suggestion
You may also use ephemeral inline volumes with deployments and other application types. See this [Kubernetes blog](https://kubernetes.io/blog/2020/01/21/csi-ephemeral-inline-volumes/) for more information and example use cases.
```

      kubectl apply -f clonePVC.yaml
      ```

## Ephemeral Inline Volumes with CSI

To use an ephemeral inline volume with the Portworx CSI driver, declare it inside of the pod spec as seen below.

```text
kind: Pod
apiVersion: v1
metadata:
  name: csi-inline-volume-app
spec:
  containers:
    - name: my-frontend
      image: busybox
      volumeMounts:
      - mountPath: "/data"
        name: my-csi-volume
      command: [ "sleep", "1000000" ]
  volumes:
    - name: my-csi-volume
      csi:
        driver: pxd.portworx.com
        volumeAttributes:
          size: "2Gi"
```

Ephemeral inline volumes may also be used with deployments and other application types. See this [Kubernetes blog](https://kubernetes.io/blog/2020/01/21/csi-ephemeral-inline-volumes/) for more information and example use cases.

### Ephemeral Inline Volumes alpha

> To enable the alpha version of this feature in Kubernetes, enable the `CSIInlineVolume` feature gate in Kubernetes 1.15:

Reviewer: Why do mention alpha here? for solutions architects and prototyping?

Author: Yes mainly - I can remove this if it adds confusion.

Author: Removed this
